pkg/controller/master/upgrade/upgrade_repo.go

```diff
@@ -303,6 +309,69 @@ func (r *UpgradeRepo) createVM(image *harvesterv1.VirtualMachineImage) (*kubevir
 	return r.h.vmClient.Create(&vm)
 }
 
+func (r *UpgradeRepo) deleteVM() error {
+	vmName := r.getVMName()
+
+	vm, err := r.h.vmCache.Get(upgradeNamespace, vmName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
```

Comment:
Not sure whether I understand correctly. If we can't find a VM here, we don't delete the image directly as the solution.

Reply:
You are correct, I'll fix it. Thanks.

Reply:
Fixed.